app/src/main/java/com/vero/tasky/auth/presentation/registration/RegistrationViewModel.kt

```diff
@@ -0,0 +1,120 @@
+package com.vero.tasky.auth.presentation.registration
+
+import androidx.annotation.StringRes
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.vero.tasky.R
+import com.vero.tasky.auth.domain.usecase.RegistrationUseCases
+import com.vero.tasky.auth.util.EmailErrorHandler
+import com.vero.tasky.auth.util.PasswordErrorHandler
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class RegistrationViewModel @Inject constructor(
+    private val registrationUseCases: RegistrationUseCases,
+    private val savedStateHandle: SavedStateHandle
+) :ViewModel() {
+
+    var state : RegistrationState by mutableStateOf(
+        savedStateHandle.get<RegistrationState>(STATE_KEY)?.copy(
+            isLoading = false
+        ) ?: RegistrationState()
+    )
+        private set
+
+    private val channel = Channel<UiRegistrationEvent>()
+    val uiEvent = channel.receiveAsFlow()
+
+    fun onEvent(event: RegistrationEvent) {
+        when(event) {
+            RegistrationEvent.OnBackPressed -> {
+                viewModelScope.launch {
+                    channel.send(UiRegistrationEvent.OnBackPressed)
+                }
+            }
```

Comment:
You don't really need to send an event to the VM just to send it back to the UI. These events that don't change any UI state can be directly processed in the UI